xc/xenops_server_xen.ml

```diff
@@ -1638,6 +1638,14 @@ module VM = struct
 							end;
 							hvm = di.Xenctrl.hvm_guest;
 							shadow_multiplier_target = shadow_multiplier_target;
+							nomigrate=Platform.is_true
```

Comment:
As discussed in person, it is more reliable to put these keys into VMextra, and set them at boot time, so that we know for sure that they have not been modified after booting the VM.

We do need to be careful that when reading an old VMextra record from disk, defaults are filled in for the missing fields, otherwise things will break in upgrade scenarios.